engines/router/missionctl/log/resultlog/kafka.go

```diff
@@ -125,33 +130,50 @@ func (l *KafkaLogger) write(turLogEntry *TuringResultLogEntry) error {
 	return nil
 }
 
+func (l *KafkaLogger) write(turLogEntry *TuringResultLogEntry) error {
+	return l.writeToKafka(
+		&turLogEntry.resultLogMessage,
+		turLogEntry.resultLogMessage.TuringReqId,
+		turLogEntry.resultLogMessage.EventTimestamp,
+	)
+}
+
+func (l *KafkaLogger) WriteUPIRouterLog(routerLog *upiv1.RouterLog) error {
```

Comment:
It feels a bit out of place. Can we instead have a separate `KafkaLogger` (e.g. `UPIKafkaLogger`) and embed the original `KafkaLogger` within it and override the `write` method?

Reply:
hmm the challenge with this, is that a UPI wrapper for each logger needs to be created, including the `nop` logger, since write is redeclared twice as the parameter is different. I can change the interface to a proto.message but all the loggers for HTTP need some refactor and type assert back to `TuringResultLog`. I prefer to do this is another MR if necessary. 

```
// TuringResultLogger is an abstraction for the underlying result logger
type TuringResultLogger interface {
	write(*TuringResultLogEntry) error
}
```

Reply:
> is that a UPI wrapper for each logger needs to be created, including the nop logger

I don't think this is needed. Since the logging orchestrator (UPIResultLogger) in itself in a separate implementation, the `logger` in the below initialization need not adhere to the `TuringResultLogger` interface anymore, right? That logger can be the `UPIKafkaLogger` which implements only the `UPILogger` interface, but makes use of the existing Kafka logger under the hood to write the given messge.

```
resultLogger, err := resultlog.InitUPIResultLogger(cfg.AppConfig.Name, logger)
```

Reply:
yep `UPIKafkaLogger` is fine, but I would need `UPINopLogger` else there be this clash. Will just add it anyway 😅 

```
// write is a nop method that satisfies the NopLogger interface
func (*NopLogger) write(turLogEntry *TuringResultLogEntry) error {
	return nil
}

// write is a nop method that satisfies the UPILogger interface
func (*NopLogger) write(routerLog *upiv1.RouterLog) error {
	return nil
}
```